test/evergreen.yml

```diff
@@ -97,6 +97,19 @@ functions:
           set -o verbose
           echo '#undef HAVE_FTRUNCATE' >> wiredtiger_config.h
     - *make_wiredtiger
+  "compile wiredtiger sanitizer":
```

Comment:
It looks like doing the similar stuff which "compile wiredtiger" does. 
How about : 
Create a task say "compile-wiredtiger-sanitizer" and call the functions 
"get project"  
"compile wiredtiger"
 set your own configure_env_vars and posix_configure_flags

Reply:
We need to compile with this particular `configure_env_vars` and `posix_configure_flags` combination for other sanitize tests in future as well. That's why I made a special function that can be called by all those tasks.

Reply:
I understand, but I prefer to use the reconf and configure options of existing functions instead of writing new. 

Something like the below task "compile-wiredtiger-sanitizer" will do the job and also it can be used in other sanitize tests.

  - name: compile-wiredtiger-sanitizer
    commands:
      - func: "get project"
      - func: "compile wiredtiger"
        vars:
          configure_env_vars: CC="clang -fsanitize=address" CFLAGS="-g -fno-omit-frame-pointer"
          posix_configure_flags: --enable-diagnostic --with-builtins=lz4,snappy,zlib

Reply:
I actually wrote a special compile task for this and made the sanitizer test to depend on that test. However, the `format.sh` script failed as we have seen before that some tests are unable to use the precompiled directories.

I wrote this special compile function as a workaround.